components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/pojo/policy/PolicyManager.java

```diff
@@ -71,11 +71,11 @@ public boolean addAutoscalePolicy(AutoscalePolicy policy) throws InvalidPolicyEx
         if (log.isInfoEnabled()) {
             log.info(String.format("Adding autoscaling policy: [id] %s", policy.getId()));
         }
-        if(StringUtils.isEmpty(policy.getId())){
+        if (StringUtils.isEmpty(policy.getId())) {
             throw new AutoScalerException("Autoscaling policy id cannot be empty");
         }
-        this.addASPolicyToInformationModel(policy);
         RegistryManager.getInstance().persistAutoscalerPolicy(policy);
+        this.addASPolicyToInformationModel(policy);
         if (log.isInfoEnabled()) {
             log.info(String.format("Autoscaling policy is added successfully: [id] %s", policy.getId()));
```

Comment:
For consistency we have been using the string format in all our loggings. But true it is not much effective for a single string.